includes/class-wc-shipping-zone.php

```diff
@@ -0,0 +1,428 @@
+		'zone_locations' => array()
+	);
+
+	/**
+	 * True when location data needs to be re-saved
+	 * @var bool
+	 */
+	private $_locations_changed = false;
+
+	/**
+	 * Constructor for zones
+	 * @param int|object $zone Zone ID to load from the DB (optional) or already queried data.
+	 */
+    public function __construct( $zone = 0 ) {
+		if ( is_numeric( $zone ) && ! empty( $zone ) ) {
+        	$this->read( $zone );
+		} elseif ( is_object( $zone ) ) {
+			$this->set_zone_id( $zone->zone_id );
+			$this->set_zone_name( $zone->zone_name );
+			$this->set_zone_order( $zone->zone_order );
+			$this->read_zone_locations( $zone->zone_id );
+		} elseif ( 0 === $zone ) {
+			$this->set_zone_name( __( 'Rest of the World', 'woocommerce' ) );
+			$this->read_zone_locations( 0 );
+		}
+    }
+
+	/**
+	 * Get ID
+	 * @return int
+	 */
+    public function get_id() {
+        return $this->get_zone_id();
+    }
+
+	/**
+	 * Get class data array
+	 * @return array
+	 */
+	public function get_data() {
+		return $this->data;
+	}
+
+	/**
+     * Insert zone into the database
+     */
+    public function create() {
+		global $wpdb;
+		$wpdb->insert( $wpdb->prefix . 'woocommerce_shipping_zones', array(
+			'zone_name'  => $this->get_zone_name(),
+			'zone_order' => $this->get_zone_order(),
+		) );
+		$this->set_zone_id( $wpdb->insert_id );
+	}
+
+	/**
+	 * Read zone.
+	 * @param int ID to read from DB
+	 */
+	public function read( $id ) {
+		global $wpdb;
+
+		if ( $zone_data = $wpdb->get_row( $wpdb->prepare( "SELECT * FROM {$wpdb->prefix}woocommerce_shipping_zones WHERE zone_id = %d LIMIT 1;", $id ) ) ) {
+			$this->set_zone_id( $zone_data->zone_id );
+			$this->set_zone_name( $zone_data->zone_name );
+			$this->set_zone_order( $zone_data->zone_order );
+			$this->read_zone_locations( $zone_data->zone_id );
+		}
+	}
+
+    /**
+     * Update zone in the database
+     */
+    public function update() {
+        global $wpdb;
+		$wpdb->update( $wpdb->prefix . 'woocommerce_shipping_zones', array(
+			'zone_name'  => $this->get_zone_name(),
+			'zone_order' => $this->get_zone_order(),
+		), array( 'zone_id' => $this->get_zone_id() ) );
+    }
+
+	/**
+	 * Delete a zone.
+	 * @since 2.6.0
+	 */
+    public function delete() {
+		if ( $this->get_id() ) {
+			global $wpdb;
+			$wpdb->delete( $wpdb->prefix . 'woocommerce_shipping_zone_methods', array( 'zone_id' => $this->get_id() ) );
+	        $wpdb->delete( $wpdb->prefix . 'woocommerce_shipping_zone_locations', array( 'zone_id' => $this->get_id() ) );
+			$wpdb->delete( $wpdb->prefix . 'woocommerce_shipping_zones', array( 'zone_id' => $this->get_id() ) );
+			WC_Cache_Helper::incr_cache_prefix( 'shipping_zones' );
+			$this->set_zone_id( 0 );
+		}
+    }
+
+	/**
+	 * Save zone data to the database.
+	 */
+	public function save() {
+		$data = array(
+			'zone_name'  => $this->get_zone_name(),
+			'zone_order' => $this->get_zone_order(),
+		);
+
+		if ( ! $this->get_zone_id() ) {
+			$this->create( $data );
```

Comment:
@mikejolley create and update don't have any parameters, so you could remove the $data array above and the parameters here.